spec/services/attribute_asserter_spec.rb

```diff
@@ -101,6 +102,18 @@
     )
     CGI.unescape ialmax_authnrequest.split('SAMLRequest').last
   end
+  let(:raw_biometric_preferred_ial_authn_request) do
+    biometric_preferred_ial_authnrequest = saml_authn_request_url(
+      overrides: {
+        issuer: sp1_issuer,
+        authn_context: [
+          Saml::Idp::Constants::IAL2_BIO_PREFERRED_AUTHN_CONTEXT_CLASSREF,
+        ],
+        authn_context_comparison: 'minimum',
+      },
+    )
+    CGI.unescape biometric_preferred_ial_authnrequest.split('SAMLRequest').last
```

Comment:
```suggestion
    UriService.params(unescape biometric_preferred_ial_authnrequest)[:SAMLRequest]
```

Reply:
Nice. I'll create a helper function as this exists in many places.

Reply:
It wasn't as straightforward as we expected. Besides the obvious difference of the string starting with '=' in the `split` approach vs the parsing approach, there were other differences in the resulting string. I don't have time to look into it now. If you're OK with it, I'll get back to investigating it after this PR is done.

Reply:
yeah that seems fine to skip for now